### ISSUE TITLE
Expects a migration to only be able to run once

### ISSUE THREAD

**Issue by henninghall** [contentful/contentful-migration#1068]
## Expected Behavior

When I run the same migration multiple times I expect it to throw an error saying that this migration has already been run.

## Actual Behavior

It does try to run the migration multiple times. It usually doesn't succeed but because of other reasons, for instance that the field already exists.

## Possible Solution

This other less popular library [contentful-migrate](https://github.com/deluan/contentful-migrate) is handling this, I think by creating a migration entry in contentful itself for each migration that has been run.

Does a solution already exist for this?

## Steps to Reproduce

1. Run any migration
2. Run the same migration again

## Environment
-   **Package Version**: 4.8.1

**Comment by torressam333** [comment]
I agree with this @henninghall , there should also be a way to rollback the migration to undo the changes that just took place.

**Comment by MosesG10** [comment]
This a problem for me as well. Not having this functionality does not give me confidence in being able to rollback effectively

**Comment by nteague22** [comment]
I put in place a manual setup for this, added a fingerprint function that I run on each migration, made a single content-type with one entry (but could be diff for diff usage) called migration-tracking, then have every migration .js file do a migration.transformEntriesForLocale that adds the name of the migration to the list on the entry.  But it throws an error if the name is in the list, so by throwing within a sync call -- it will auto hook into the migrations cli dry-run verification

**Comment by nteague22** [comment]
so the entity is just a symbol list and a name, I did make from the mgmt api, so I could make the tracker name able to be an easy lookup -- but I was modelling it after the efcore migrations Microsoft does, would be nice to spawn an up/down system though -- to allow for rolling them back

**Comment by sbonami** [comment]
🧟 

I've had success in other projects using https://github.com/sequelize/umzug, which provides a generic, agnostic framework for running migrations. I can't imagine it would be too much work to build a Contentful-specific store or for users to BYOD.
